src/main/java/akka/stream/contrib/FileTailSource.java

```diff
@@ -0,0 +1,174 @@
+            }
+          });
+        }
+
+        @Override
+        public void preStart() {
+          chunkCallback = createAsyncCallback((tryInteger) -> {
+            if (tryInteger.isSuccess()) {
+              int readBytes = tryInteger.get();
+              if (readBytes > 0) {
+                buffer.flip();
+                push(out, ByteString.fromByteBuffer(buffer));
+                position += readBytes;
+                buffer.clear();
+              } else {
+                // hit end, try again in a while
+                scheduleOnce("poll", pollingInterval);
+              }
+
+            } else {
+              failStage(tryInteger.failed().get());
+            }
+
+          });
+        }
+
+        @Override
+        public void onTimer(Object timerKey) {
+          doPull();
+        }
+
+
+        private void doPull() {
+          channel.read(buffer, position, chunkCallback, completionHandler);
+        }
+
+        @Override
+        public void postStop() {
+          try {
+            if (channel.isOpen()) channel.close();
+          } catch(Exception ex) {
+            // Remove when #21168 is fixed
+            throw new RuntimeException(ex);
+          }
+        }
+      };
+
+    } catch (Exception ex) {
+      // remove when #21168 is fixed
+      throw new RuntimeException(ex);
+    }
+  }
+
+
+  // factory methods
+
+  /**
+   * Java API:
+   *
+   * Read the entire contents of a file, and then when the end is reached, keep reading
+   * newly appended data. Like the unix command `tail -f`.
+   *
+   * Aborting the stage can be done by combining with a [[akka.stream.KillSwitch]]
+   *
+   * @param path a file path to tail
+   * @param maxChunkSize The max emitted size of the `ByteString`s
+   * @param startingPosition Offset into the file to start reading
+   * @param pollingInterval When the end has been reached, look for new content with this interval
+   */
+  public static Source<ByteString, NotUsed> create(Path path, int maxChunkSize, long startingPosition, FiniteDuration pollingInterval) {
+    return Source.fromGraph(new FileTailSource(path, maxChunkSize, startingPosition, pollingInterval));
+  }
+
+  /**
+   * Scala API:
+   *
+   * Read the entire contents of a file, and then when the end is reached, keep reading
+   * newly appended data. Like the unix command `tail -f`.
+   *
+   * Aborting the stage can be done by combining with a [[akka.stream.KillSwitch]]
+   *
+   * @param path a file path to tail
+   * @param maxChunkSize The max emitted size of the `ByteString`s
+   * @param startingPosition Offset into the file to start reading
+   * @param pollingInterval When the end has been reached, look for new content with this interval
+   */
+  public static akka.stream.scaladsl.Source<ByteString, NotUsed> apply(Path path, int maxChunkSize, long startingPosition, FiniteDuration pollingInterval) {
+    return create(path, maxChunkSize, startingPosition, pollingInterval).asScala();
+  }
```

Comment:
This doesn't really work. Maybe put a corresponding object in the Scala sources for it.